File: Problems/AddTwoNumbers.php
/**
 * Definition for a singly-linked list.
 * class ListNode {
 *     public $val = 0;
 *     public $next = null;
 *     function __construct($val) { $this->val = $val; }
 * }
 */
class Solution {

    /**
     * @param ListNode $l1
     * @param ListNode $l2
     * @return ListNode
     */
    function addTwoNumbers($l1, $l2) {
        $result=new ListNode($this->res + $l1->val + $l2->val);
        if($this->res=intval($result->val >9)){
            $result->val -=10;
        }
        $result->next=(!$this->res && is_null($l1->next) && is_null($l2->next) )?null:$this->addTwoNumbers($l1->next,$l2->next);
        return $result;
    }
}

File: Problems/AddTwoNumbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        
        if not l1:
            return l2
        elif not l2:
            return l1
        
        result = iterator = ListNode(0)
        decimal, sum = 0, 0
                       
        while l1 or l2:
            if l1:
                sum += l1.val
                l1 = l1.next
            if l2:
                sum += l2.val
                l2 = l2.next
            
            sum += decimal
            
            iterator.next = ListNode(sum % 10)
            decimal = sum // 10
            iterator = iterator.next
            sum = 0
        
        if(decimal):
            iterator.next = ListNode(decimal)
            
        return result.next
